### DISCUSSION TITLE
How to signout and clear auth when Apollo GraphQL token is invalid or it has expired

### DISCUSSION THREAD

**Discussion by JavierMartinz** [vercel/next.js#19647] [Help]
What would it be the best way to clear the `NextAuth.js` session when trying to reach the backend (Apollo GraphQL) and it returns a `401` because the token has expired or is invalid?

I thought about an `errorLink` and `signOut`, but as far as I know `signOut` cannot be used server side at `getServerSideProps`, but only client-side.

What is the recommended way to do so? Is there any other way to implement a middleware to take care of that scenario?

This would be the proof of concept of the `errorLink` I'm trying to implement, the code is caught in that if but I can't use `signOut()` as it's only available client-side.

```js
const errorLink = onError(({ graphQLErrors }) => {
   if (graphQLErrors?.[0]?.message === 'Unauthenticated.') {
    // signOut(); & redirect here??? How to signOut from here???
  }
});

function createApolloClient(session) {
  return new ApolloClient({
    cache: new InMemoryCache(),
    ssrMode: typeof window === 'undefined',
    link: from([
      errorLink,
      createUploadLink({
        uri: GRAPHQL_URI,
        credentials: 'same-origin',
        headers: { Authorization: session?.accessToken ? `Bearer ${session.accessToken}` : '' },
      }),
    ]),
  });
}
```

**Reference**
https://next-auth.js.org/getting-started/client#signout

**Comment by waptik** [comment]
check this [next-auth#rest-api](https://next-auth.js.org/getting-started/rest-api#post-apiauthsignout) out. You can use fetch api directly inside `getServerSideProps `or `getStaticProps`.
